src/python/pants/init/engine_initializer.py

```diff
@@ -296,17 +297,17 @@ def setup_legacy_graph_extended(
     pants_ignore_patterns,
     local_store_dir,
     build_file_imports_behavior,
-    options_bootstrapper,
-    build_configuration,
-    build_root=None,
-    native=None,
-    glob_match_error_behavior=None,
+    options_bootstrapper: OptionsBootstrapper,
+    build_configuration: BuildConfiguration,
+    build_root: Optional[str] = None,
+    native: Optional[Native] = None,
+    glob_match_error_behavior: Optional[GlobMatchErrorBehavior] = None,
```

Comment:
Type annotations to complex functions with many arguments = *chef's kiss*!

Reply:
Yes 😍